### PR TITLE
Input Component Updates

### PR DESCRIPTION
**Pull request by sammartinez**
_Description of changes:_
* Added the `name` prop to the Input component
* Added additional unit tests for default values on the props

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-examples/amplify/.config/local-env-info.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "projectPath": "/Users/jranz/amplify/amplify-js/packages/amplify-ui-components/src/components/amplify-examples",
+  "projectPath": "/Users/samlmar/Code/amplify/amplify-js/packages/amplify-ui-components/src/components/amplify-examples",
```
Comment (jordanranz) [inline packages/amplify-ui-components/src/components/amplify-examples/amplify/.config/local-env-info.json]: I think we are missing a `.gitignore` in the amplify-examples directory. Can you add one. Staging samples has an example of one generated by Amplify CLI

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,4 +1,19 @@
 {
 	"editor.formatOnSave": true,
-	"editor.insertSpaces": false
+	"editor.insertSpaces": false,
```
Comment (jordanranz) [inline .vscode/settings.json]: This stuff should be ignored

File: packages/amplify-ui-components/src/components/amplify-examples/amplify/.config/local-env-info.json (hunk 1 above)
Comment (sammartinez) [reply packages/amplify-ui-components/src/components/amplify-examples/amplify/.config/local-env-info.json]: yup yup

File: .vscode/settings.json (hunk 2 above)
Comment (sammartinez) [reply .vscode/settings.json]: Its strange cause the top level `.gitignore` has this. Ill remove it

File: packages/amplify-ui-components/src/components/amplify-input/amplify-input.spec.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { newSpecPage } from '@stencil/core/testing';
-import { AmplifyInput } from './amplify-input'
+import { AmplifyInput } from './amplify-input';
```
Comment (jordanranz) [inline packages/amplify-ui-components/src/components/amplify-input/amplify-input.spec.ts]: Can you add a snapshot test for adding the name prop

Comment (sammartinez) [reply packages/amplify-ui-components/src/components/amplify-input/amplify-input.spec.ts]: sure

**Pull request merged**